# Expo 55 Canary iOS Build Fix for Xcode 16.4
# This patch fixes Swift 6 concurrency errors when building with Xcode 16.4
#
# Apply this patch after upgrading to Expo 55 canary
# The Swift file patches need to be reapplied after yarn install

diff --git a/ios/Podfile b/ios/Podfile
index 1234567..abcdefg 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -50,6 +50,23 @@ target 'igniteexpo54to55' do
   )

   post_install do |installer|
     react_native_post_install(
       installer,
       config[:reactNativePath],
       :mac_catalyst_enabled => false,
       :ccache_enabled => ccache_enabled?(podfile_properties),
     )
+
+    # Disable Swift 6 concurrency checking (Xcode 16+) to fix Expo 55 canary build errors
+    # The ExpoModulesCore podspec sets swift_version = '6.0', we need to override it
+    installer.pods_project.targets.each do |target|
+      target.build_configurations.each do |config|
+        # Override Swift 6 from podspec, use 5.10 which supports @MainActor
+        config.build_settings['SWIFT_VERSION'] = '5.10'
+        # Explicitly disable strict concurrency
+        config.build_settings.delete('SWIFT_STRICT_CONCURRENCY')
+        # Add flags to disable concurrency warnings/errors
+        other_swift_flags = config.build_settings['OTHER_SWIFT_FLAGS'] ||= ['$(inherited)']
+        # Remove any existing concurrency flags
+        other_swift_flags.delete_if { |flag| flag.to_s.include?('concurrency') }
+      end
+    end
+
+    # Also update the pods project-level settings
+    installer.pods_project.build_configurations.each do |config|
+      config.build_settings['SWIFT_VERSION'] = '5.10'
+      config.build_settings.delete('SWIFT_STRICT_CONCURRENCY')
+    end
   end
 end

diff --git a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
index 1234567..abcdefg 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
@@ -42,7 +42,7 @@ extension ExpoSwiftUI {
   /**
    A hosting view that renders a SwiftUI view inside the UIKit view hierarchy.
    */
-  public final class HostingView<Props: ViewProps, ContentView: View<Props>>: ExpoView, @MainActor AnyExpoSwiftUIHostingView {
+  public final class HostingView<Props: ViewProps, ContentView: View<Props>>: ExpoView, AnyExpoSwiftUIHostingView {
     /**
      Props object that stores all the props for this particular view.
      It's an environment object that is observed by the content view.

diff --git a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
index 2345678..bcdefgh 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
@@ -121,7 +121,7 @@ extension ConcurrentFunctionDefinition: ViewDefinitionFunctionElement {
   public typealias ViewType = FirstArgType
 }

-extension UIView: @MainActor AnyArgument {
+extension UIView: AnyArgument {
   public static func getDynamicType() -> AnyDynamicType {
     return DynamicViewType(innerType: Self.self)
   }